src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -28,64 +28,69 @@ internal ModelOperationsCatalog(IHostEnvironment env)
         {
-            _env.CheckValue(model, nameof(model));
+            _env.CheckValue(loader, nameof(loader));
+            _env.CheckValueOrNull(model);
             _env.CheckValue(stream, nameof(stream));
 
+            // For the sake of consistency of this API specifically, when called upon we save any transformer
+            // in a single element transformer chain.
+            var chainedModel = model == null ? null : new TransformerChain<ITransformer>(model);
+            var compositeLoader = new CompositeDataLoader<TSource, ITransformer>(loader, chainedModel);
+
             using (var rep = RepositoryWriter.CreateNew(stream))
             {
-                ModelSaveContext.SaveModel(rep, model, null);
+                ModelSaveContext.SaveModel(rep, compositeLoader, null);
                 rep.Commit();
             }
         }
 
-        /// <summary>
-        /// Save the model to the file.
-        /// </summary>
-        /// <param name="model">The trained model to be saved.</param>
-        /// <param name="filePath">Path where model should be saved.</param>
-        public void Save<TSource>(IDataLoader<TSource> model, string filePath)
-        {
-            using (var stream = File.Create(filePath))
-                Save(model, stream);
-        }
-
-        /// <summary>
-        /// Save a transformer model and the loader used to create its input data to the stream.
-        /// </summary>
-        /// <param name="loader">The loader that was used to create data to train the model</param>
-        /// <param name="model">The trained model to be saved</param>
-        /// <param name="stream">A writeable, seekable stream to save to.</param>
-        public void Save<TSource>(IDataLoader<TSource> loader, ITransformer model, Stream stream) =>
-            Save(new CompositeDataLoader<TSource, ITransformer>(loader, new TransformerChain<ITransformer>(model)), stream);
-
         /// <summary>
         /// Save a transformer model and the loader used to create its input data to the file.
         /// </summary>
-        /// <param name="loader">The loader that was used to create data to train the model</param>
-        /// <param name="model">The trained model to be saved</param>
+        /// <param name="model">The trained model to be saved. Note that this can be <see langword="null"/>, as a shorthand
+        /// for an empty data set. Upon loading with <see cref="LoadWithDataLoader(Stream, out IDataLoader{IMultiStreamSource})"/>
+        /// the returned valued will be an empty <see cref="TransformerChain{TLastTransformer}"/>.</param>
+        /// <param name="loader">The loader that was used to create data to train the model.</param>
         /// <param name="filePath">Path where model should be saved.</param>
-        public void Save<TSource>(IDataLoader<TSource> loader, ITransformer model, string filePath)
+        public void Save<TSource>(ITransformer model, IDataLoader<TSource> loader, string filePath)
```

Comment:
I'm also curious about whether we agree that what is effectively the descriptor of input to the model should come after the model itself. The API had been inconsistent, in that in one save method we had loader/model (so input to model came first), whereas in another we had model/schema (so input to model came second).

In all cases I feel like the load methods should always just return `ITransform`, so maybe just always putting `ITransform` 
front-and-center for saving was probably the best approach, so I resolved the inconsistency that way, but I don't insist on it.

I kept it as a separate commit in case we wanted to go the other way, maybe... so in that case, the schema.